OpenRA.Mods.Common/Traits/World/Pathfinder/PathSearch.cs

```diff
@@ -0,0 +1,197 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+
+using OpenRA.Primitives;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	public sealed class PathSearch : BasePathSearch
+	{
+		public override CellLayer<CellInfo> CellInfo { get; protected set; }
+
+		public override IEnumerable<Pair<CPos, int>> Considered
+		{
+			get { return considered; }
+		}
+
+		LinkedList<Pair<CPos, int>> considered;
```

Comment:
That's interesting! Usually I'd expect List to be faster since it's much more cache friendly.

Also as a note, List has an amortized constant time for appending an item at the end. When the internal array is resized, its size is multiplied by some constant factor. This means you trade some excess space for faster appends since you don't need to resize every time.

If profiling says LinkedList is faster, I'll believe it. My only thought might be that creating a LinkedListNode for each item might make a bit more work for the GC but I doubt it'll be substantial.